Classifier: Move resize observer into DefaultLayout to aid responsive styling

goplayoutside3 opened the pull request:
## Package
lib-classifier

## Linked Issue and/or Talk Post
Initially started in https://github.com/zooniverse/front-end-monorepo/pull/3927
Working toward responsive styling in the Image Toolbar.

## Describe your changes

A resize observer was initially added in Flipbook's controls to aid in styling for a responsive layout. The subject viewer grid area becomes narrow on mobile screens and on 700px - 900px screens (still 2-column layout). 

I moved the resize observer code to DefaultLayout so that ImageToolbar, MetaTools, and other subject viewer components such as flipbook and video controls can adapt to the same "breakpoint". Note that this is _not_ a viewport breakpoint, the resize observer is watching the subject viewer grid area `div`'s width, and setting the SubjectViewerStore's `viewerWidth` attribute.

in the future, this code could be replaced by [@container](https://developer.mozilla.org/en-US/docs/Web/CSS/@container) css queries, but that feature is yet to be fully implemented in Firefox.

Further updates to stories and tests:
    - Added a DefaultLayout story
    - Converted DefaultLayout.spec to RTL
    - Removed Layout.spec as it's not a valuable test
    - Added a SubjectViewerFactory for use in mocks
    - Updated Flipbook's "small styling" story to use a subject viewer mock

## How to Review

Run `yarn bootstrap` with this branch and view [Wildwatch Burrowing Owl](https://local.zooniverse.org:3000/projects/sandiegozooglobal/wildwatch-burrowing-owl/classify/workflow/21141?env=production) or a [test project](https://local.zooniverse.org:3000/projects/goplayoutside3/fem-flipbook-viewer/classify/workflow/22532?env=production) locally. Resize the browser, or view it on a mobile phone. When the subject viewer grid area is less than 550px, the flipbook controls' styling adapts.

In storybook, view the DefaultLayout story or Flipbook's "small styling"vs "default" stories.

The changes in this PR will be used in my next PR revamping the Image Toolbar with responsive styling (working in branch `responsive-image-toolbar`).

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

eatyourgreens commented:
I think this might break if the classifier uses a layout other than the default layout. It would probably be safer to couple the subject viewer to the generic `Layout` interface, rather than tie it to a specific classifier layout. 

For example, will this still work once we add the fullscreen layout to the classifier, or will the viewer still try to size itself as if it’s constrained inside a two column grid?

goplayoutside3 commented:
@eatyourgreens I wasn't aware that more layouts were in the pipeline. There are three components I've been wrestling with in terms of responsiveness in layout shifts: flipbook controls, the image toolbar, and metatools. An important note - viewport width as measured by [withResponsiveLayout](https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-react-components/src/helpers/withResponsiveContext) isn’t as useful to these components because their design is much more linked to the shifting layout around them rather than viewport width.

The image toolbar is my main concern, as it doesn’t have any options for responsive styling at the moment. The toolbar must also respond to users zooming their browser window. I just updated #4008 to use css rules for responsive styling instead of listening for container width.

I also removed this PR's code changes from my follow-up PR's (ImageToolbar #4008, Metatools #4021). Flipbook controls can always have a resize detector directly in its component file, and MetaTools as a flexbox can use different cues than listening for container width.

If those ideas for the image toolbar and metatools seem like a better longterm solution, I'll close this PR.
